CLN: tests/extension/json/test_json.py typefix

lukasbk opened the pull request:
The error is due to the fact that the builtins.staticmethod type defined in extension.base.base.BaseExtensionTests does not match the function type in extension.json.test_json.BaseJSON. I don't see a possibility for a proper type hint fix without changing the structure of the code. One way would be to replace the call of staticmethod by function definitions that call the corresponding methods.
Here I propose an easier suppression of the error using type: Any.
What do you think?

- [ ] xref #28926
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

lukasbk committed:
CLN: fix mypy errors in pandas/tests/extension/test_numpy.py #28926

lukasbk committed:
Applied suggestions, PR also fixes errors in test_sparse.py

lukasbk committed:
Adjusted imports

lukasbk committed:
CLN: added type hint for mypy to test_coercion.py

lukasbk committed:
Merge remote-tracking branch 'upstream/master' into test_coercion_typefix

lukasbk committed:
Fixed mypy error in test_json.py

The error is due to the fact that the builtins.staticmethod type defined
 in extension.base.base.BaseExtensionTests does not match the function
type in extension.json.test_json.BaseJSON. I don't see a possibility for
a type hint fix without changing the structure of the code. One way
would be to replace the call of staticmethod by function definitions
that call the corresponding methods.
Here I propose an easier suppression of the error using type: Any.
What do you think?

lukasbk committed:
reformatting

simonjayhawkins reviewed (changes requested):
@lukasbk Thanks for the PR.

This PR also has the changes from #28990, can you keep these separate.

also see #28988 re discussion on solving `Definition of "assert_series_equal" in base class "BaseJSON" is incompatible with definition in base class "BaseExtensionTests"` errors.

lukasbk committed:
Revert "CLN: added type hint for mypy to test_coercion.py"

This reverts commit 217adc84dc69045c566ec2c510ae11e1d531bbac.

lukasbk committed:
Merge branch 'master' into test_json_typefix

jbrockmendel commented:
@lukasbk can you rebase

lukasbk committed:
Matching arguments of assert_..._equal methods

pep8speaks commented:
Hello @lukasbk! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/tests/extension/base/base.py`](https://github.com/pandas-dev/pandas/blob/acf78f9269baf8ff2d67c386183c1d6f952a47c3/pandas/tests/extension/base/base.py):

> [Line 22:70](https://github.com/pandas-dev/pandas/blob/acf78f9269baf8ff2d67c386183c1d6f952a47c3/pandas/tests/extension/base/base.py#L22): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

lukasbk committed:
Reformatting

simonjayhawkins committed:
Merge branch 'master' into test_json_typefix

simonjayhawkins commented on pandas/tests/extension/base/base.py:
```diff
@@ -2,8 +2,18 @@
 class BaseExtensionTests:
+    @staticmethod
+    def assert_equal(left, right, **kwargs):
+        return tm.assert_equal(left, right, **kwargs)
```
don't need the return

simonjayhawkins commented on pandas/tests/extension/base/base.py:
```diff
@@ -2,8 +2,18 @@
-    assert_extension_array_equal = staticmethod(tm.assert_extension_array_equal)
+    @staticmethod
+    def assert_series_equal(left, right, **kwargs):
+        return tm.assert_series_equal(left, right, **kwargs)
```
same

simonjayhawkins commented on pandas/tests/extension/base/base.py:
```diff
@@ -2,8 +2,18 @@
+
+    @staticmethod
+    def assert_frame_equal(left, right, **kwargs):
+        return tm.assert_frame_equal(left, right, **kwargs)
```
same

simonjayhawkins commented on pandas/tests/extension/base/base.py:
```diff
@@ -2,8 +2,18 @@
+
+    @staticmethod
+    def assert_extension_array_equal(left, right, **kwargs):
+        return tm.assert_extension_array_equal(left, right, **kwargs)
```
same

lukasbk committed:
Removed returns

lukasbk committed:
fix setup.cfg merge conflict

lukasbk committed:
Merge branch 'test_json_typefix' of github.com:lukasbk/pandas into test_json_typefix

WillAyd commented on pandas/tests/extension/base/base.py:
```diff
@@ -2,8 +2,18 @@
 
 
 class BaseExtensionTests:
+    @staticmethod
```
Can you get rid of these methods altogether and just import / use as function within tests?

simonjayhawkins replied:
I don't think so, since some tests need to override these

WillAyd replied:
Hmm OK - so was this change required? Looks like using `tm.assert_*` in the tests now?

WillAyd commented:
@lukasbk looks like a merge conflict; can you fix up and address comments?

simonjayhawkins commented on pandas/tests/extension/base/base.py:
```diff
@@ -2,8 +2,18 @@
 
 class BaseExtensionTests:
+    @staticmethod
+    def assert_equal(left, right, **kwargs):
```
AFAICT it's only `assert_series_equal` and `assert_frame_equal` that currently need to be overridden in specific extension array tests, so `assert_equal` and `assert_extension_array_equal` don't need to be changed at this time.

lukasbk committed:
Resolve merge in setup.cfg and reduce base.py

assert_equal and assert_extension_array_equal were not needed to be overwritten. Instead they can be directly accessed from testing.

lukasbk committed:
Fix stata files

lukasbk committed:
Change more references to tm

jreback commented:
looks ok to me. I am bit concerned that we are now changing back to using ``tm`` as we *do* want the ability to override on these tests, but I guess that mypy will check this now.

jreback commented:
can you rebase

WillAyd commented:
@lukasbk can you fix up the merge conflict? I think we can get this in if so

lukasbk committed:
Merge master

datapythonista commented:
@lukasbk the CI didn't pass, and this is conflicting again.

Do you mind merging master again, making sure the CI passes, and pinging us to get this merged quickly after that please? Thanks!

WillAyd commented:
@lukasbk can you fix up merge conflict? I think this was close

simonjayhawkins commented:
closing as stale. cc @SaturnFromTitan
